File: java/src/entities/AnagraficaMansione.java

package entities;

/**
 * Classe che implementa una mansione all'interno di una struttura
 * @author emanuelegargiulo
 * @author giandomenicoizzo
 */
public class AnagraficaMansione {
    
    private String codiceFiscaleAnagrafica;
    private String password;
    private String nomeStruttura;
    private String codiceFiscaleProprietario;
    private int tipoMansione;

    /**
     * Costrutture vuoto
     */
    public AnagraficaMansione() {
    }
    
    /**
     * Restituisce il codice fiscale dell'anagrafica che svolge la mansione
     * @return codice fiscale
     */
    public String getCodiceFiscaleAnagrafica() {
        return codiceFiscaleAnagrafica;
    }

    /**
     * Setta il codice fiscale dell'anagrafica che svolge la mansione
     * @param codiceFiscaleAnagrafica
     */
    public void setCodiceFiscaleAnagrafica(String codiceFiscaleAnagrafica) {
        this.codiceFiscaleAnagrafica = codiceFiscaleAnagrafica;
    }

    /**
     * Restituisce la password per accedere al programma scelta dall'anagrafica
     * @return password
     */
    public String getPassword() {
        return password;
    }

    /**
     * Setta la password scelta dall'utente
     * @param password
     */
    public void setPassword(String password) {
        this.password = password;
    }

    /**
     * Restituisce il nome della struttura in cui viene svolta la mansione
     * @return nome struttura
     */
    public String getNomeStruttura() {
        return nomeStruttura;
    }

    /**
     * Setta il nome della struttura in cui viene svolta la mansione
     * @param nomeStruttura
     */
    public void setNomeStruttura(String nomeStruttura) {
        this.nomeStruttura = nomeStruttura;
    }

    /**
     * Restituisce il tipo di mansione svolta
     * @return tipo mansione
     */
    public int getTipoMansione() {
        return tipoMansione;
    }

    /**
     * Setta il tipo di mansione svolta
     * @param tipoMansione
     */
    public void setTipoMansione(int tipoMansione) {
        this.tipoMansione = tipoMansione;
    }

    /**
     * Restituisce il codice fiscale del proprietario della struttura
     * @return codice fiscale proprietario
     */
    public String getCodiceFiscaleProprietario() {
        return codiceFiscaleProprietario;
    }

    /**
     * Setta il codice fiscale del proprietario della struttura
     * @param codiceFiscaleProprietario
     */
    public void setCodiceFiscaleProprietario(String codiceFiscaleProprietario) {
        this.codiceFiscaleProprietario = codiceFiscaleProprietario;
    }
    
}

File: deliverables/Programma/server/entity/AnagraficaMansione.php
<?php

class AnagraficaMansione {
    
    private $codiceFiscaleAnagrafica;
    private $password;
    private $nomeStruttura;
    private $codiceFiscaleProprietario;
    private $tipoMansione;
    
    function getCodiceFiscaleAnagrafica() {
        return $this->codiceFiscaleAnagrafica;
    }

    function getPassword() {
        return $this->password;
    }

    function getNomeStruttura() {
        return $this->nomeStruttura;
    }

    function getTipoMansione() {
        return $this->tipoMansione;
    }

    function setCodiceFiscaleAnagrafica($codiceFiscaleAnagrafica) {
        $this->codiceFiscaleAnagrafica = $codiceFiscaleAnagrafica;
        return $this;
    }

    function setPassword($password) {
        $this->password = $password;
        return $this;
    }

    function setNomeStruttura($nomeStruttura) {
        $this->nomeStruttura = $nomeStruttura;
        return $this;
    }

    function setTipoMansione($tipoMansione) {
        $this->tipoMansione = $tipoMansione;
        return $this;
    }

    function getCodiceFiscaleProprietario() {
        return $this->codiceFiscaleProprietario;
    }

    function setCodiceFiscaleProprietario($codiceFiscaleProprietario) {
        $this->codiceFiscaleProprietario = $codiceFiscaleProprietario;
        return $this;
    }

            
    function toArray() {
        return array(
            "codiceFiscaleAnagrafica" => $this->codiceFiscaleAnagrafica,
            "nomeStruttura" => $this->nomeStruttura,
            "codiceFiscaleProprietario" => $this->codiceFiscaleProprietario,
            "tipoMansione" => $this->tipoMansione,
            "password" => $this->password
        );
    }

}